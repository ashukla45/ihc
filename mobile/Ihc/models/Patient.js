import {stringDate} from '../util/Date';
export default class Patient {
  // Insert any class methods here
  get drugUpdates() {
    const drugToUpdates = {};
    for (var update of this.medications) {
      if (update.name in drugToUpdates) {
        drugToUpdates[update.name].push(update);
      } else{
        drugToUpdates[update.name] = [update];
      }
    }
    return drugToUpdates;
  }

  get age() {
    const strBirthday = this.birthday;
    const today = stringDate(new Date());
    const birthdayYear = parseInt(strBirthday.slice(0,4)); // Get the year
    const birthdayMonthDay= parseInt(strBirthday.slice(4));
    const todayYear = parseInt(today.slice(0,4));
    const todayMonthDay = parseInt(today.slice(4));

    let age = todayYear - birthdayYear;
    // If hasn't passed their birthday this year, then subtract
    if(birthdayMonthDay < todayMonthDay) {
      age--;
    }
    return age;
  }

  /* Returns {
   *   weights: [[ageInMonths, weight (kg)],...]},
   *   heights: [[ageInMonths, height (cm)],...]}
   * }
   */
  get growthChartData() {
    const triages = this.triages;
    const weights = [];
    const heights = [];
    triages.forEach( triage => {
      const ageInMonths = triage.age * 12; // Rounds off to whole year, should be fine
      weights.push([ageInMonths, triage.weight]);
      heights.push([ageInMonths, triage.height]);
    });
    return {weights, heights};
  }

  get isMale() {
    return this.gender === 1;
  }

  static fullName(patient) {
    return `${patient.firstName} ${patient.fatherName} ${patient.motherName}`;
  }

  // To be used as primary key
  static makeKey(patient) {
    const str = `${patient.firstName}&${patient.fatherName}&${patient.motherName}` +
      `&${patient.birthday}`;
    return str;
  }

  static extractFromForm(form) {
    const patient = Object.assign({}, form);
    patient.birthday = stringDate(form.birthday);
    patient.key = Patient.makeKey(patient);
    patient.needToUpload = false;
    if(form.newPatient) {
      // 1 is male, 2 is female
      patient.gender = form.gender === 'Male' ? 1 : 2;
    }
    return patient;
  }

  // Can pass in parameters to override defaults, mostly useful for tests
  static getInstance(lastUpdated = new Date().getTime(),
    key = 'firstname&father&mother&20000101', firstName = 'firstname',
    fatherName = 'father', motherName = 'mother', birthday = '20000101',
    gender = 1, phone = null, motherHeight = 100, fatherHeight = 100,
    medications = [], soaps = [], triages = [], statuses = []) {
    return {
      key, firstName, fatherName, motherName, birthday, gender, phone, motherHeight,
      fatherHeight, medications, soaps, triages, statuses, lastUpdated
    };
  }
}

Patient.schema = {
  name: 'Patient',
  primaryKey: 'key',
  properties: {
    key: 'string',
    firstName: 'string',
    fatherName: 'string', // last name
    motherName: 'string', // last name
    birthday: 'string',
    gender: 'int', // 1 = boy, 2 = girl, 0 = undefined
    phone: 'string?',
    motherHeight: 'double?',
    fatherHeight: 'double?',
    medications: 'DrugUpdate[]',
    soaps: 'Soap[]',
    triages: 'Triage[]',
    statuses: 'Status[]',
    lastUpdated: 'int', // timestamp
    needToUpload: 'bool'
  }
};
