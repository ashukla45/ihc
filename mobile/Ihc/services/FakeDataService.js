/* eslint-disable */
// This file should hold skeletons for the function calls from RealDataService.
// Implementation is not required because they should be stubbed out during
// tests
export function getPatient(patientKey) {
  return {};
}

export function getTriage(patientKey, date) {
  return {};
}

export function getSoap(patientKey, date) {
  return {};
}

export function getPatients(param) {
  return []; // Return whatever sample data you want
}

export function getUpdates(param) {
  return []; // Return whatever sample data you want
}

/*
 * active_only is true if we only want the statuses of patients that are
 * currently checked in
 */
export function getPatientSelectRows() {
  const arr =
    [
      newFakePatientSelectRow('Jose'),
      newFakePatientSelectRow('Bob'),
      newFakePatientSelectRow('Matt'),
      newFakePatientSelectRow('Coco'),
      newFakePatientSelectRow('Brandon'),
      newFakePatientSelectRow('Evan'),
      newFakePatientSelectRow('Sarah'),
      newFakePatientSelectRow('Daniel'),
      newFakePatientSelectRow('Neil'),
      newFakePatientSelectRow('William')
    ];

  const columnOrder = ['name', 'birthday', 'checkin_time', 'triage_completed',
    'doctor_completed', 'pharmacy_completed', 'notes'];
  const toReturn = createMatrix(arr, columnOrder);

  return new Promise((resolve, reject) => {
    setTimeout(resolve, 100, toReturn);
  });
}

export function getMedicationUpdates() {
  const arr = [
    newFakeMedicationUpdate('Tylenol'),
    newFakeMedicationUpdate('Tylenol'),
    newFakeMedicationUpdate('Advil'),
    newFakeMedicationUpdate('Aspirin'),
    newFakeMedicationUpdate('Aspirin'),
    newFakeMedicationUpdate('Aspirin'),
    newFakeMedicationUpdate('Cold meds'),
  ];

  return new Promise((resolve, reject) => {
    setTimeout(resolve, 100, arr);
  });
}

const medsCount = 0;
function newFakeMedicationUpdate(name) {
  medsCount = ++medsCount % 3;

  return {
    name: name,
    date: `2018020${medsCount}`,
    dose: '10mg',
    frequency: 'bid',
    duration: '1 month',
    notes: 'yes'
  };
}

const numFakePatients = 0;
// Currently ignore triage/doctor/pharmacy params
function newFakePatientSelectRow(name, triage = false, doctor = false,
  pharm = false, notes = '') {
  numFakePatients++;
  return {
    name: name,
    birthday: '20180201',
    checkin_time: new Date().getTime(),
    triage_completed: numFakePatients % 2 === 0 ? new Date().getTime() : null,
    doctor_completed: numFakePatients % 4 === 0 ? new Date().getTime() : null,
    pharmacy_completed: numFakePatients % 8 === 0 ? new Date().getTime() : null,
    notes: notes
  };
}

function createMatrix(data, columnOrder) {
  return data.map((obj) => columnOrder.map( (key) => obj[key] ));
}
/* eslint-enable */
