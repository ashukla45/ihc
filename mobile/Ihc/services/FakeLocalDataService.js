/* eslint-disable */
// This file should hold skeletons for the function calls from RealDataService.
// Implementation is not required because they should be stubbed out during
// tests
export function createPatient(patient) {
}

export function signinPatient(patientForm) {
}

export function updateStatus(patientKey, strDate, field, value) {
}

export function createDrugUpdate(update) {
}

export function updateSoap(update) {
}

export function getSoap(patientKey, strDate) {
}

export function updateTriage(update) {
}

export function getTriage(patientKey, strDate) {
}

export function getPatient(patientKey) {
}

export function getPatientsToUpload() {
}

export function lastSynced() {
}

export function handleDownloadedPatients(patients) {
}

/*
 * active_only is true if we only want the statuses of patients that are
 * currently checked in
 */
export function getPatientSelectRows() {
  const arr =
    [
      newFakePatientSelectRow('Jose'),
      newFakePatientSelectRow('Bob'),
      newFakePatientSelectRow('Matt'),
      newFakePatientSelectRow('Coco'),
      newFakePatientSelectRow('Brandon'),
      newFakePatientSelectRow('Evan'),
      newFakePatientSelectRow('Sarah'),
      newFakePatientSelectRow('Daniel'),
      newFakePatientSelectRow('Neil'),
      newFakePatientSelectRow('William')
    ];

  const columnOrder = ['name', 'birthday', 'checkin_time', 'triage_completed',
    'doctor_completed', 'pharmacy_completed', 'notes'];
  const toReturn = createMatrix(arr, columnOrder);

  return new Promise((resolve, reject) => {
    setTimeout(resolve, 100, toReturn);
  });
}

export function getMedicationUpdates() {
  const arr = [
    newFakeMedicationUpdate('Tylenol'),
    newFakeMedicationUpdate('Tylenol'),
    newFakeMedicationUpdate('Advil'),
    newFakeMedicationUpdate('Aspirin'),
    newFakeMedicationUpdate('Aspirin'),
    newFakeMedicationUpdate('Aspirin'),
    newFakeMedicationUpdate('Cold meds'),
  ];

  return new Promise((resolve, reject) => {
    setTimeout(resolve, 100, arr);
  });
}

const medsCount = 0;
function newFakeMedicationUpdate(name) {
  medsCount = ++medsCount % 3;

  return {
    name: name,
    date: `2018020${medsCount}`,
    dose: '10mg',
    frequency: 'bid',
    duration: '1 month',
    notes: 'yes'
  };
}

const numFakePatients = 0;
// Currently ignore triage/doctor/pharmacy params
function newFakePatientSelectRow(name, triage = false, doctor = false,
  pharm = false, notes = '') {
  numFakePatients++;
  return {
    name: name,
    birthday: '20180201',
    checkin_time: new Date().getTime(),
    triage_completed: numFakePatients % 2 === 0 ? new Date().getTime() : null,
    doctor_completed: numFakePatients % 4 === 0 ? new Date().getTime() : null,
    pharmacy_completed: numFakePatients % 8 === 0 ? new Date().getTime() : null,
    notes: notes
  };
}

function createMatrix(data, columnOrder) {
  return data.map((obj) => columnOrder.map( (key) => obj[key] ));
}
/* eslint-enable */
