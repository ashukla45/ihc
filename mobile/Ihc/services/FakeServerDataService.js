/* eslint-disable */
// This file should hold skeletons for the function calls from RealDataService.
// Implementation is not required because they should be stubbed out during
// tests

export function createPatient(patient) {
  return new Promise((res,rej) => {
    setTimeout(() => {
      res(true);
      // rej(new Error("Fake error in FakeServerDataService"));
    }, 500);
  });
}

export function updateStatus(statusObj) {
  return new Promise((res,rej) => {
    setTimeout(() => {
      res(true);
    }, 500);
  });
}

// Server endpoint: put /patient/:key/drugUpdate
export function createDrugUpdate(update) {
  return new Promise((res,rej) => {
    setTimeout(() => {
      res(true);
    }, 500);
  });
}

// Server endpoint: put /patient/:key/soap/:date
export function updateSoap(update) {
  return new Promise((res,rej) => {
    setTimeout(() => {
      res(true);
    }, 500);
  });
}

// Server endpoint: get /patient/:key/soap/:date
// Returns SOAP object if it exists, or undefined if not
export function getSoap(patientKey, strDate) {
}

// Server endpoint: put /patient/:key/triage/:date
export function updateTriage(update) {
  return new Promise((res,rej) => {
    setTimeout(() => {
      res(true);
    }, 500);
  });
}

// Server endpoint: get /patient/:key/triage/:date
// Returns Triage object if it exists, or undefined if not
export function getTriage(patientKey, strDate) {
}

// Server endpoint: get /patient/:key
export function getPatient(patientKey) {
}

// Server endpoint: get /patient/:key/drugUpdates
export function getMedicationUpdates(patientKey) {
}

// Server endpoint: get /patients/statuses/:date
// Return the statuses of the patients that are active and for this date
export function getActiveStatuses() {
}

// Server endpoint: post /patients
export function updatePatients(patients) {
  return new Promise((res,rej) => {
    setTimeout(() => {
      res(true);
    }, 500);
  });
}

// Server endpoint: get /patients/:timestamp
export function getUpdatedPatients(lastSynced) {
}
/* eslint-enable */
