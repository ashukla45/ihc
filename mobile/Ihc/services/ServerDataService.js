// This file should handle all the fetch() calls to the Express server
// Requires Promises because dealing with server requests

import config from '../config.json';
import {convertPatientForServer, convertStatusForServer} from '../util/Realm';
// Must set the fetchUrl to the server's IP Address and Port
const fetchUrl = config.fetchUrl;

// Server endpoint: post /patient
export function createPatient(patient) {
  return fetch(fetchUrl + '/patient', {
    method: 'POST',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({
      patient: patient
    })
  }).then(response => response.json())
    .then(json => {
      // status is false if the Network connection went through but there was
      // some kind of error when processing the request.
      if (!json.status) {
        throw new Error(json.error);
      }

      return Promise.resolve(true);
    }).catch(err => {
      return Promise.reject(err);
    });
}

// Server endpoint: put /patient/:key/status/:date
export function updateStatus(statusObj) {
  const copy = convertStatusForServer(statusObj);
  return fetch(fetchUrl + `/patient/${statusObj.patientKey}/status/${statusObj.date}`, {
    method: 'PUT',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({
      status: copy
    })
  }).then(response => response.json())
    .then(json => {
      // status is false if the Network connection went through but there was
      // some kind of error when processing the request. Throwing an error here
      // will lead to patient.needToUpload being marked as true
      if (!json.status) {
        throw new Error(json.error);
      }

      return Promise.resolve(true);
    }).catch(err => {
      return Promise.reject(err);
    });
}

// Server endpoint: put /patient/:key/drugUpdate
export function createDrugUpdate(update) {
}

// Server endpoint: put /patient/:key/soap/:date
export function updateSoap(update) {
}

// Server endpoint: get /patient/:key/soap/:date
// Returns SOAP object if it exists, or undefined if not
export function getSoap(patientKey, strDate) {
}

// Server endpoint: put /patient/:key/triage/:date
export function updateTriage(update) {
}

// Server endpoint: get /patient/:key/triage/:date
// Returns Triage object if it exists, or undefined if not
export function getTriage(patientKey, strDate) {
}

// Server endpoint: get /patient/:key
export function getPatient(patientKey) {
}

// Server endpoint: get /patient/:key/drugUpdates
export function getMedicationUpdates(patientKey) {
}

// Server endpoint: get /patients/statuses/:date
// Return the statuses of the patients that are active and for this date
export function getActiveStatuses() {
}

// Server endpoint: post /patients
export function updatePatients(patients) {
  const patientsCopy = patients.map( patient => convertPatientForServer(patient) );
  return fetch(fetchUrl + '/patients/', {
    method: 'PUT',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({
      patients: patientsCopy
    }),
  }).then(() => {
    return Promise.resolve(true);
  }).catch(err => {
    return Promise.reject(err);
  });
}

// Server endpoint: get /patients/:timestamp
export function getUpdatedPatients(lastSynced) {
  return fetch(fetchUrl + '/patients/' + lastSynced)
    .then(response => response.json())
    .then(json => {
      const patients = json.patients;
      return Promise.resolve(patients);
    }).catch(err => {
      return Promise.reject(err);
    });
}
