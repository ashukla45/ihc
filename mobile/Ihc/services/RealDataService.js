// This file should handle all the fetch() calls to the Express server
// and the local database calls to the Realm DB
// All lastUpdated fields on mobile-side should be handled within these
// functions

import Patient from '../models/Patient';
import Status from '../models/Status';
import Soap from '../models/Soap';
import Triage from '../models/Triage';
import DrugUpdate from '../models/DrugUpdate';
import Settings from '../models/Settings';

import {createPatientHelper} from './functions/CreatePatient';
import {signinPatientHelper} from './functions/SigninPatient';
import {updateStatusHelper} from './functions/UpdateStatus';
import {updateSoapHelper} from './functions/UpdateSoap';
import {createDrugUpdateHelper} from './functions/CreateDrugUpdate';
import {getSoapHelper} from './functions/GetSoap';
import {getTriageHelper} from './functions/GetTriage';
import {getPatientHelper} from './functions/GetPatient';
import {getPatientSelectRowsHelper} from './functions/GetPatientSelectRows';
import {updateTriageHelper} from './functions/UpdateTriage';
import {getMedicationUpdatesHelper} from './functions/GetMedicationUpdates';
import {uploadUpdatesHelper} from './functions/UploadUpdates';
import {downloadUpdatesHelper} from './functions/DownloadUpdates';

import Realm from 'realm';
import config from '../config.json';

// Must set the fetchUrl to the server's IP Address and Port
const fetchUrl = config.fetchUrl;

const realm = new Realm({
  schema: [Patient, Status, Soap, Triage, DrugUpdate, Settings],
  deleteRealmIfMigrationNeeded: true, // TODO: delete when done with dev
});

// Server endpoint: post /patient
export function createPatient(patient) {
  return createPatientHelper(patient, realm, fetchUrl);
}

// Server endpoint: put /patient/:key/status/:date
// Check that patient exists locally, and if so then create a status object for them
export function signinPatient(patientForm) {
  return signinPatientHelper(patientForm, realm, fetchUrl);
}

// Server endpoint: put /patient/:key/status/:date
// field: Any of fields from Status.schema
// value: should match the type that the Status.schema says
export function updateStatus(patientKey, strDate, field, value) {
  return updateStatusHelper(patientKey, strDate, field, value, realm, fetchUrl);
}

// Server endpoint: put /patient/:key/drugUpdate
export function createDrugUpdate(update) {
  return createDrugUpdateHelper(update, realm, fetchUrl);
}

// Server endpoint: put /patient/:key/soap/:date
export function updateSoap(update) {
  return updateSoapHelper(update, realm, fetchUrl);
}

// Server endpoint: get /patient/:key/soap/:date
// Returns SOAP object if it exists, or undefined if not
export function getSoap(patientKey, strDate) {
  return getSoapHelper(patientKey, strDate, realm, fetchUrl);
}

// Server endpoint: put /patient/:key/triage/:date
export function updateTriage(update) {
  return updateTriageHelper(update, realm, fetchUrl);
}

// Server endpoint: get /patient/:key/triage/:date
// Returns Triage object if it exists, or undefined if not
export function getTriage(patientKey, strDate) {
  return getTriageHelper(patientKey, strDate, realm, fetchUrl);
}

// Server endpoint: get /patient/:key
export function getPatient(patientKey) {
  return getPatientHelper(patientKey, realm, fetchUrl);
}

// Server endpoint: get /patient/:key/drugUpdates
export function getMedicationUpdates(patientKey) {
  return getMedicationUpdatesHelper(patientKey, realm, fetchUrl);
}

// Server endpoint: get /patients/statuses
// Return the statuses of the patients that are active and for this date
export function getPatientSelectRows() {
  return getPatientSelectRowsHelper(realm, fetchUrl);
}

// Server endpoint: post /patients
export function uploadUpdates() {
  return uploadUpdatesHelper(realm, fetchUrl);
}

// Server endpoint: get /patients/:timestamp
export function downloadUpdates() {
  return downloadUpdatesHelper(realm, fetchUrl);
}
