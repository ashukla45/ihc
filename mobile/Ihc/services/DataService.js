/* global */
// Returns either fakeLocalData or RealData depending on config file
// Config file:
// "FakeDataServices": "true" or "false"
import * as fakeLocalData from '../services/FakeLocalDataService';
import * as localDataService from '../services/LocalDataService';

import * as fakeServerData from '../services/FakeServerDataService';
import * as serverDataService from '../services/ServerDataService';

import config from '../config.json';

// Dont use the Fakes in tests, so that we can test the LocalDataService itself,
// but should stub out its methods for the other tests
let localData = (config.fakeLocalDataService === 'true') ? fakeLocalData : localDataService;

// If we are testing the server data service, then use the real one
let serverData = (config.fakeServerDataService === 'true' && config.testingServerDataService == 'false')
  ? fakeServerData : serverDataService;

export {localData, serverData};
