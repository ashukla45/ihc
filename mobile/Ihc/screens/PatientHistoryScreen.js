import React, { Component } from 'react';
import {
  StyleSheet,
  TouchableOpacity,
  Text,
  View,
} from 'react-native';
import { Col, Grid } from 'react-native-easy-grid';
import {localData} from '../services/DataService';
import {formatDate} from '../util/Date';
import {shortDate} from '../util/Date';
import Container from '../components/Container';

/* TODO: 
 * Make changes in behavior for the cases that a soap form is submitted, 
 * but not a triage form and vice versa
*/

export default class PatientHistoryScreen extends Component<{}> {
  /*
   * Expects:
   *  {
   *    name: string, patient's name (for convenience)
   *    patientKey: string
   *  }
   */
  constructor(props) {
    super(props);
    this.state = {
      patient: null,
      loading: false,
      errorMsg: null
    };
  }

  loadPatient = () => {
    this.setState({ loading: true });
    try {
      const patient = localData.getPatient(this.props.patientKey);
      this.setState({ patient: patient, loading: false });
    } catch(err) {
      this.setState({ errorMsg: err.message, loading: false });
    }
  }

  componentDidMount() {
    this.loadPatient();
  }

  goToSoap(date) {
  
    this.props.navigator.push({
      screen: 'Ihc.SoapScreen',
      title: 'Back to patient',
      passProps: { name: this.props.name, patientKey: this.props.patientKey, todayDate: date }
    });
  }

  goToTriage(date) { 
    this.props.navigator.push({
      screen: 'Ihc.TriageScreen',
      title: 'Back to patient',
      passProps: { patientKey: this.props.patientKey, todayDate: date }
    });
  }

  render() {
    if (this.state.patient == null) {
      return (
        <Container loading={this.state.loading}
          errorMsg={this.state.errorMsg} >
          
          <Text style={styles.title}>
            Previous Visits
          </Text>
          <Text>Patient doesnt exist...</Text>
        </Container>
      );
    }

    return (
      <Container loading={this.state.loading}
        errorMsg={this.state.errorMsg} >

        <Text style={styles.title}>
          Previous Visits
        </Text>

        <View style={styles.gridContainer}>
          <Grid>
            <Col style={styles.col}>
              {this.state.patient.soaps.map( (soap, i) => 
                <Text key={i} style={styles.dateContainer}>{formatDate(new Date(shortDate(soap.date)))}</Text> )}
            </Col>

            <Col style={styles.col}>
              {this.state.patient.soaps.map( (soap, i) => 
                <TouchableOpacity key={i} style={styles.buttonContainer}
                  onPress={() => this.goToSoap(soap.date)}>
                  <Text style={styles.button}>SOAP</Text>
                </TouchableOpacity> ) }
            </Col>

            <Col style={styles.col}>
              {this.state.patient.soaps.map( (soap, i) => 
                <TouchableOpacity key={i} style={styles.buttonContainer}
                  onPress={() => this.goToTriage(soap.date)}>
                  <Text style={styles.button}>Triage</Text>
                </TouchableOpacity> ) }
            </Col>
          </Grid>
        </View>
      </Container>
    );
  }
}
const styles = StyleSheet.create({
  gridContainer: {
    flex: 1,
    maxWidth: '80%',
    alignItems: 'center',
  },
  col: {
    alignItems: 'center',
  },
  title: {
    fontSize: 20,
    textAlign: 'center',
    margin: 5,
  },
  buttonContainer: {
    width: 150,
    margin: 10,
    padding: 8,
    elevation: 4,
    borderRadius: 2,
    backgroundColor: '#2196F3',
  },
  dateContainer: {
    width: 150,
    margin: 10,
    padding: 8,
    elevation: 4,
  },
  button: {
    fontWeight: '500',
    color: '#fefefe',
    textAlign: 'center',
  }
});
