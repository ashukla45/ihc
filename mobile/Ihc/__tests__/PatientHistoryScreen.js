import 'react-native';
import React from 'react';
import PatientHistoryScreen from '../screens/PatientHistoryScreen';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import sinon from 'sinon';
import {localData} from '../services/DataService';
import Patient from '../models/Patient';

it('renders correctly', () => {
  sinon.stub(localData, 'getPatient').returns(Patient.getInstance());

  const json = renderer.create(
    <PatientHistoryScreen todayDate = '20180101' />
  ).toJSON();
  expect(json).toMatchSnapshot();
});
