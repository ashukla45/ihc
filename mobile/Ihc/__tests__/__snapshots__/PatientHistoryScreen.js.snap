// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F5FCFF",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "flex": 0,
        "justifyContent": "center",
        "padding": 20,
      }
    }
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "margin": 5,
            "textAlign": "center",
          }
        }
      >
        Previous Visits
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "maxWidth": "80%",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "column",
              }
            }
          />
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "column",
              }
            }
          />
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "column",
              }
            }
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F5FCFF",
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "red",
          "textAlign": "center",
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "green",
          "textAlign": "center",
        }
      }
    />
  </View>
</View>
`;
